io/io/src/TBufferFile.cxx

```diff
@@ -3297,7 +3297,7 @@ Int_t TBufferFile::ReadClassBuffer(const TClass *cl, void *pointer, Int_t versio
    /// The StreamerInfo should exist at this point.
 
    else {
-      R__LOCKGUARD(gInterpreterMutex);
+      R__READ_LOCKGUARD(ROOT::gCoreMutex);
```

Comment:
How likely will we deadlock here, @pcanal ?

Reply:
There is no (new) risk there.  By design the Read and Write lock are working well togethere (a thread that hold a read lock and then request a write lock, first realease the read lock and waiting until all other thread have release the Read lock either by finishing they critical section or asking for the write lock).

I am more worried about race condition.   Currently the `StreamerInfo` is published via `RegisterStreamerInfo` which is called before the end of the critical section.  However since the check is done through the ReadCritical section and there is a test on IsCompiled which is updated at the last line of Build, I am guessing it is likely okay.